pyvista/core/filters/data_set.py

```diff
@@ -2695,11 +2702,17 @@ def glyph(
         else:
             raise ValueError(f"Invalid color mode '{color_mode}'")
 
+        if vector_mode == 'vector':
+            alg.SetVectorModeToUseVector()
+        elif vector_mode == 'normal':
+            alg.SetVectorModeToUseNormal()
+        else:
+            raise ValueError(f"Invalid vector mode '{vector_mode}'")
```

Comment:
Thank you for your contribution! Could you add some test code for this code? Other than that, it's perfect.

Reply:
@tkoyama010 I was a little bit quick writing this fix... when I started making tests it is not as simple as I thought. 

Looking at the vtk code in more detail it the `SetVectorModeToUseNormal` doesn't change the way the glyphs are interpreted rather it changes to using an attribute on the vtk object inNormals... See [here](https://github.com/Kitware/VTK/blob/1fe1cd1f467d432aa18b45f0b8a8ef278dfb8b5d/Filters/Generic/vtkGenericGlyph3DFilter.cxx#L176).

What confused me is the different behaviour between how normal is interpreted as a property of a pyvista geometry class and how that normal is used by the glyph filter...

`import pyvista as pv

p1 = pv.PolyData(np.array([[0, 0, 0]]))
p1["norm"] = np.array([[1, 0, 0]])
p2 = pv.PolyData(np.array([[1, 1, 1]]))
p2["norm"] = np.array([[0, 1, 0]])
p3 = pv.PolyData(np.array([[2, 2, 2]]))
p3["norm"] = np.array([[0, 0, 1]])

disk1 = pv.Disc(center=[0, 0, 0], normal=[1, 0, 0], inner=0)
disk2 = pv.Disc(center=[1, 1, 1], normal=[0, 1, 0], inner=0)
disk3 = pv.Disc(center=[2, 2, 2], normal=[0, 0, 1], inner=0)
geom = pv.Disc()
p1 = p1.glyph(
    geom=geom,
    orient="norm",
)
p2 = p2.glyph(
    geom=geom,
    orient="norm",
)
p3 = p3.glyph(
    geom=geom,
    orient="norm",
)
view = pv.Plotter()
view.add_mesh(disk1, name="disk1", color="red")
view.add_mesh(disk2, name="disk2", color="green")
view.add_mesh(disk3, name="disk3", color="blue")
view.add_mesh(p1, name="p1", color="red")
view.add_mesh(p2, name="p2", color="green")
view.add_mesh(p3, name="p3", color="blue")
view.show()`


![image](https://github.com/user-attachments/assets/c80816c2-f3ac-47df-9da5-4786d8991d62)


If we use a glyph like arrow
import pyvista as pv

p1 = pv.PolyData(np.array([[0, 0, 0]]))
p1["norm"] = np.array([[1, 0, 0]])
p2 = pv.PolyData(np.array([[1, 1, 1]]))
p2["norm"] = np.array([[0, 1, 0]])
p3 = pv.PolyData(np.array([[2, 2, 2]]))
p3["norm"] = np.array([[0, 0, 1]])

disk1 = pv.Arrow(start=[0, 0, 0], direction=[1, 0, 0], )
disk2 = pv.Arrow(start=[1, 1, 1], direction=[0, 1, 0], )
disk3 = pv.Arrow(start=[2, 2, 2], direction=[0, 0, 1], )
geom = pv.Arrow()
p1 = p1.glyph(
    geom=geom,
    orient="norm",
)
p2 = p2.glyph(
    geom=geom,
    orient="norm",
)
p3 = p3.glyph(
    geom=geom,
    orient="norm",
)
view = Loop3DView()
view.add_mesh(disk1, name="disk1", color="red")
view.add_mesh(disk2, name="disk2", color="green")
view.add_mesh(disk3, name="disk3", color="blue")
view.add_mesh(p1, name="p1", color="red")
view.add_mesh(p2, name="p2", color="green")
view.add_mesh(p3, name="p3", color="blue")
view.show()
![image](https://github.com/user-attachments/assets/5b127dca-609d-4b91-962f-6288a8c112d1)

I found that if I rotate the geometry by 90 degrees before passing it to the glyph method it will give me the results I expect. 
`import pyvista as pv

p1 = pv.PolyData(np.array([[0, 0, 0]]))
p1["norm"] = np.array([[1, 0, 0]])
p2 = pv.PolyData(np.array([[1, 1, 1]]))
p2["norm"] = np.array([[0, 1, 0]])
p3 = pv.PolyData(np.array([[2, 2, 2]]))
p3["norm"] = np.array([[0, 0, 1]])

disk1 = pv.Disc(center=[0, 0, 0], normal=[1, 0, 0], inner=0)
disk2 = pv.Disc(center=[1, 1, 1], normal=[0, 1, 0], inner=0)
disk3 = pv.Disc(center=[2, 2, 2], normal=[0, 0, 1], inner=0)
geom = pv.Disc()
geom = geom.rotate_y(90)
p1 = p1.glyph(
    geom=geom,
    orient="norm",
)
p2 = p2.glyph(
    geom=geom,
    orient="norm",
)
p3 = p3.glyph(
    geom=geom,
    orient="norm",
)
view = pv.Plotter()
view.add_mesh(disk1, name="disk1", color="red")
view.add_mesh(disk2, name="disk2", color="green")
view.add_mesh(disk3, name="disk3", color="blue")
view.add_mesh(p1, name="p1", color="red")
view.add_mesh(p2, name="p2", color="green")
view.add_mesh(p3, name="p3", color="blue")
view.add_arrows(np.array([[0,0,0],[1,1,1],[2,2,2]]),np.array([[1,0,0],[0,1,0],[0,0,1]]),color='black')

view.show()`

I noticed the same issue with the rectangle glyph and parametric ellipsoid depending on the aspect ratio of the ellipsoid.  I can modify this PR to just rotate the geometry by 90 degrees in the y direction if "normals" is chosen but I am not sure if it would be better to use a different name.